test/integration/Quota.test.ts

```diff
@@ -0,0 +1,177 @@
+async function performSimplePUTWithLength(path: string, length: number): Promise<Response> {
+  return fetch(
+    path,
+    {
+      method: 'PUT',
+      headers: {
+        'content-type': 'text/plain',
+      },
+      body: Array.from({ length: length + 1 }).join('A'),
+    },
+  );
+}
+
+/** Registers two test pods on the server matching the 'baseUrl' */
+async function registerTestPods(baseUrl: string): Promise<void> {
+  await fetch(`${baseUrl}idp/register/`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/x-www-form-urlencoded',
+      'content-length': '120',
+    },
+    body: 'createWebId=on&webId=&register=on&createPod=on&' +
+      'podName=abel&email=abel%40example.ai&password=t&confirmPassword=t&submit=',
+  });
+  await fetch(`${baseUrl}idp/register/`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/x-www-form-urlencoded',
+      'content-length': '124',
+    },
+    body: 'createWebId=on&webId=&register=on&createPod=on&' +
+      'podName=arthur&email=arthur%40example.ai&password=t&confirmPassword=t&submit=',
+  });
+}
+
+describe('A server with', (): void => {
+  /** Test the general functionality of the server using pod quota */
+  describe('pod quota enabled', (): void => {
+    const port = getPort('PodQuota');
+    const baseUrl = `http://localhost:${port}/`;
+    const rootFilePath = getTestFolder('quota-pod');
+
+    let app: App;
+
+    beforeAll(async(): Promise<void> => {
+      // Calculate the allowed quota depending on file system used
+      const folderSizeTest = await fsPromises.stat(process.cwd());
+      const size = (folderSizeTest.size * 2) + 4000;
+
+      const instances = await instantiateFromConfig(
+        'urn:solid-server:test:Instances',
+        getTestConfigPath('quota-pod.json'),
+        {
+          ...getDefaultVariables(port, baseUrl),
+          'urn:solid-server:default:variable:rootFilePath': rootFilePath,
+          'urn:solid-server:default:variable:PodQuota': size,
+        },
+      ) as Record<string, any>;
+      ({ app } = instances);
+      await app.start();
+
+      // Initialize 2 pods
+      await registerTestPods(baseUrl);
+    });
+
+    afterAll(async(): Promise<void> => {
+      await app.stop();
+      await removeFolder(rootFilePath);
+    });
+
+    // Test quota in the first pod
+    it('should return a 413 when the quota is exceeded during write.', async(): Promise<void> => {
+      const testFile1 = `${baseUrl}abel/profile/test1.txt`;
+      const testFile2 = `${baseUrl}abel/test2.txt`;
+
+      const response1 = performSimplePUTWithLength(testFile1, 500);
+      await expect(response1).resolves.toBeDefined();
+      expect((await response1).status).toEqual(201);
+
+      const response2 = performSimplePUTWithLength(testFile2, 3000);
+      await expect(response2).resolves.toBeDefined();
+      expect((await response2).status).toEqual(413);
```

Comment:
I just realized that recursively deleting them through HTTP would be quite a hassle since you would need to authenticate for that. An easier solution would be to just delete them directly on disk and only keep the container .meta since you need that one to identify it as a pod. Or you could even manually just create such a container instead of registering.